app/utils/entity_utils.rb

```diff
@@ -145,11 +143,15 @@ def model_to_hash(model)
     transform_with: -> (transformer, v) { transformer.call(v) }
   }
 
-  def validator_or_transformer(k)
+  def categorice_spec(k)
```

Comment:
My suggestion for renaming is `categorice_spec` -> `spec_category`. You group by spec category and you get the category of a spec by calling this function.